### ISSUE TITLE
Implement the `duration` format validator

### ISSUE THREAD

**Issue by Stranger6667** [Stranger6667/jsonschema-rs#265]

**Comment by ZFLloyd** [comment]
I have taken a look to see if I could implement this but this might end-up being more difficult than expected.

The json-schema spec `duration` format is based on ISO8601:
- https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3.1
- https://datatracker.ietf.org/doc/html/rfc3339#appendix-A

From what I can see, `jsonschema-rs` relies on the `chrono` for any time related format validation, but it is currently missing a way to parse and validate ISO8601 Durations.
There was some work on this at some point but it seems it has not moved forward in a while ([PR Discussion on the implementation in chrono here](https://github.com/chronotope/chrono/pull/245))

I've tried to look for an already battle-tested parser for ISO8601 Durations, but I did not manage to find one thus far. So implementing the `duration` format would also entail implementing a proper parser which could prove tricky since I expect `jsonschema-rs` should be able to deal with untrusted input.

Since I spent some time researching this I wanted to share it here, which can hopefully clarify the issue a little.
